documentation/docs/libraries/java/android_development.md

```diff
@@ -0,0 +1,217 @@
+:::
+
+## Setup by compiling
+
+Compiling an Android app comes with the added task of building for multiple different device architectures.
+In this tutorial we mention the followint 2 variables:
+`$ARCH` and `$TARGET`.
+
+These variables are used to generate correct binaries during cross compilation and linking.
+
+#### List of target Devices
+`$ARCH` and `$TARGET` related to each other in the following manner: 
+- `armeabi-v7a` -> `armv7-linux-androideabi`
+- `arm64-v8a` -> `aarch64-linux-android`
+- `x86` -> `i686-linux-android`
+- `x86_64` -> `x86_64-linux-android`
+
+When building an app, you as a developer must decide for how many architectures you make an APK for. Binaries need to be generated for each architecture.
+
+The android-app example has a [build.gradle](../../../../bindings/java/examples/android-app/build.gradle) file that shows the enabled list in a variable called `archTriplets`. You can disable and enable them, as long as you have at least one. (This file will also automatically compile our binaries when we run gradle. Feel free to use it in your project!)
+
+We will use `archTriplets` for the enabled list of device targets during this tutorial.
+
+## Prerequisite
+
+For this setup we use `$ANDROID_NDK_HOME` for the location of your NDK, wether you use Android studio or manual compilation
+
+- Dependencies indicated in the [Getting started](getting_started.md) Prerequisite section
+- Android NDK or Android Studio with NDK installed (If you extract make sure to make it executable `chmod -R +x android-ndk-VERSION` )
+
+In order to cross compile the binaries for Android; we need the following target toolchains: (All of the enabled `archTriplets`) 
+```
+rustup target add \
+    armv7-linux-androideabi \
+    aarch64-linux-android \
+    i686-linux-android \
+    x86_64-linux-android
+```
+
+## 1. Generating the java files
+
+### 1.1 Compiling a binary
+In order to generate the Java source files; we need to run cargo manually once. 
+
+This step will require you to run `cargo build --release --target=$TARGET` in `iota.rs/bindings/java`.
+
+Replace `$TARGET` with one of the enabled targets inside `archTriplets`.
+
+:::info
+This step is actually intentionally more complicated, as you can just compile using `cargo build --release`. However running it with the correct target saves time during compilation later on.
+:::
+
+### 1.2 Creating the jar
+Afterwards, you need to run `./gradlew jar` in `iota.rs/bindings/java` in order to generate the jar file.
+
+The jar will be found at `iota.rs/bindings/java/native/build/libs/native.jar`
+
+## 2. Build the app
+
+Building the actual app can be done through two different ways. Using Android Studio and by manual linking. 
+
+The following 2 Sections describe both methods. 
+
+#### Cross compile note
+
+In order to build on windows, we need to add android triplets to our VCPKG and use that during compilation. 
+[TODO]
```

Comment:
Should this stay?

Reply:
yes i havent had t ime to delve into VCPKG, i have never used it unfortunately